tests/Doctrine/MongoDB/Tests/Query/BuilderTest.php

```diff
@@ -261,6 +260,25 @@ public function testUnsetField()
         $this->assertEquals($expected, $qb->getNewObj());
     }
 
+    public function testSetOnInsert()
+    {
+        $qb = $this->getTestQueryBuilder()
+            ->update()
+            ->upsert()
+            ->field('username')->equals('boo')
+            ->field('createDate')->setOnInsert(new \MongoDate());
+
+        $expected = array(
+            'username' => 'boo'
+        );
+        $this->assertEquals($expected, $qb->getQueryArray());
+
+        $expected = array('$setOnInsert' => array(
+            'createDate' => new \MongoDate()
```

Comment:
Isn't this making test fragile? I think that storing `new \MongoDate()` in variable and using it in qb and here could be better